README.md

```diff
@@ -1,11 +1,153 @@
 
-For developers, see [CONTRIBUTING.md](./CONTRIBUTING.md)
+> ### Want a framework that supports federated learning on the edge, in desktop browsers, integrates well with mobile apps, is performant, and preserves privacy? Welcome to XayNet, written entirely in Rust!
+
+## Making federated learning easy for developers
+Frameworks for machine learning - including those expressly for federated learning - exist already. These frameworks typically require the use of specific machine learning technology - for example tensorflow - or facilitate federated learning of cross-silo use cases - for example in collaborative learning across a limited number of hospitals.
+We want to give developers more freedom of choice and abilities in the creation of federated learning software. By doing this, we hope to also increase the pace and scope of adoption of federated learning in practice.
+Concretely, we provide developers with:
+- [**My AI tools**] the flexibility to use the machine-learning frameworks and tools of their choice,
+- [**My app dev tools**] the ability to integrate federated learning into apps written in Dart, Flutter or other languages of choice,
+- [**“Federated learning” everywhere**] the ability to run federated learning everywhere - be it desktop browsers, smartphones or micro-controllers,
+- [**“Federated learning” inside**] a simple integration means of making an AI application ready for federated learning,
+- [**Privacy by design**] a communication protocol for federated learning that scales, is secure, and preserves the privacy of participating devices.
+
+## The case for writing this framework in Rust
+
+Rust has definitely potential as a host language for machine learning itself. But, above, we already insisted on giving developers freedom of choice here. Hence, we selected Rust for other reasons.
+Our framework for federated learning is not a framework for machine learning as such. Rather, it supports the *federation* of machine learning that takes place on possibly heterogenuous devices and where use cases involve *many* such devices.
+The programming language in which this framework is written should therefore give us strong support for the following:
+- [**Compiles and runs “everywhere”**] The language should *not* require its own runtime and code should compile on a wide range of devices.
+- [**Memory and Concurreny Safety**] Code that compiles should be both memory safe and free of data races.
+- [**Secure communication**] State of the art cryptography should be available in vetted implementations.
+- [**Asynchronous communication**] Abstractions for asynchronous communication should exist that make federated learning scale.
+- [**Fast and functional**] The language should offer functional abstractions but also compile code into fast executables.
+Rust is one of the very few choices of modern programming languages that meet these requirements:
+- Its concepts of *Ownership* and *Borrowing* make it both memory and concurreny safe.
```

Comment:
Maybe `thread` instead of `concurreny`?
wdyt @finiteprods @janpetschexain

Reply:
since this intro text is the official wording we got, i wouldn't change to much except for typos. and i just noticed that `concurrency` is missing a `c` there.

Reply:
I've updated the typo on "concurreny", but as for the choice of words between `thread` and `concurrency`, I'd leave to @finiteprods, @janpetschexain and @mrahuth! :)

Reply:
2¢: _concurrency-safe_ isn't commonly used compared to _thread-safe_... but the meaning is clear enough!

Reply:
Thanks for these comments! It the readership is only developers then "thread" may be better than "concurrency". But this readme may also be read by non-developers, journalists for example. One compromise may be to say that concurrency-safe means, technically, thread-safe, then all audiences would be ok with that.

Reply:
How about, then:
> - Its concepts of *Ownership* and *Borrowing* make it both memory and thread-safe (hence avoiding potential concurrency issues).

?